src/go/rpk/pkg/cli/cmd/common/common.go

```diff
@@ -124,22 +124,30 @@ func DeduceBrokers(
 			return []string{"127.0.0.1:9092"}
 		}
 
+		if len(conf.Redpanda.KafkaApi) == 0 {
+			log.Trace(
+				"The config file contains no kafka listeners." +
+					" Empty redpanda.kafka_api.",
+			)
+			return []string{}
+		}
+
 		// Add the seed servers' Kafka addrs.
 		if len(conf.Redpanda.SeedServers) > 0 {
 			for _, b := range conf.Redpanda.SeedServers {
 				addr := fmt.Sprintf(
 					"%s:%d",
 					b.Host.Address,
-					conf.Redpanda.KafkaApi.Port,
+					conf.Redpanda.KafkaApi[0].Port,
 				)
 				bs = append(bs, addr)
 			}
 		}
 		// Add the current node's Kafka addr.
 		selfAddr := fmt.Sprintf(
 			"%s:%d",
-			conf.Redpanda.KafkaApi.Address,
-			conf.Redpanda.KafkaApi.Port,
+			conf.Redpanda.KafkaApi[0].Address,
+			conf.Redpanda.KafkaApi[0].Port,
```

Comment:
is it always guranteed to be the 0th ?

Reply:
I thought any listener would be ok, but now that I think about it, there might be listeners that only resolve from the other nodes. E.g. an entry in /etc/hosts that doesn't exist in the current node. That said, the only way to know for sure whether it resolves would be to initialize a client here.